Restructures scaling.py into a scaling module.

rmlarose opened the pull request:
With the addition of parameter scaling (#288), it makes sense to now have a `scaling` module instead of single file. Additionally, the old `scaling.py` was long enough to warrant a module on its own.

The structure is now

```
zne/
  scaling/
    conversions.py
    folding.py
    parameter.py
```

and one can import the five implemented noise scaling methods (`fold_left`, `fold_right`, `fold_random`, `fold_global`, `scale_parameters`) from `mitiq.zne.scaling`.

nathanshammah commented:
I proposed using `scaling` as a sub-package in #293, but a comment by @andreamari (+ @rmlarose) pointed out the issues coming with deep nesting, which we agreed upon, e.g., ``mitiq.zne.scaling.parameter.function`` is quite nested. I think this is too complex. Also, why would `conversions` fit into scaling?

rmlarose commented:
> I proposed using `scaling` as a sub-package in #293, but a comment by @andreamari (+ @rmlarose) pointed out the issues coming with deep nesting, which we agreed upon, e.g., `mitiq.zne.scaling.parameter.function` is quite nested. I think this is too complex. Also, why would `conversions` fit into scaling?

The nesting is the same as before, namely `mitiq.zne.scaling.scaling_function`. where `scaling_function` is one of

1. `fold_gates_from_left`
1. `fold_gates_from_right`
1. `fold_gates_at_random`
1. `fold_global`
1. `scale_parameters`

We can make them visible from `zne` if we want.

Conversions are needed for scaling because all noise scaling functions act on Cirq circuits but input `QPROGRAM`s. If (when) we need conversions for other components of the library, we can certainly move this module, but for now this is the only place its used.

andreamari reviewed (commented):
Thanks @rmlarose! 
Soon or later we may want `convert_to_mitiq` and `convert_from_mitiq` to be outside the `scaling` module. E.g. directly in `utils.py,` or in a new `conversions.py` file placed in the root folder.

Not sure if it is better to do it now in this PR or later. For the rest, LGTM.

nathanshammah commented:
I'm moving this to Milestone 6.

willzeng commented:
I'd vote for `conversions.py` to exist above the `/scaling` module

rmlarose commented:
Moved conversions.py out of scaling, + added tests for conversions. Not sure why coverage is failing (any ideas @karalekas?) but ready for re-review.

andreamari reviewed (approved):
LGTM! Thanks @rmlarose. 
The codcov error is unclear also to me.

karalekas commented:
@rmlarose the problem is that somehow even though the "diff" is fully covered, the overall coverage percentage for the repository is going down. I'll take a quick look -- it might be the `__init__.py` file.

karalekas commented:
@rmlarose It's unclear to me -- I would just ignore the failure and go ahead and merge.

rmlarose commented:
Thanks @karalekas. Yeah ironically coverage went down even though I added tests, wanted to make sure I wan't missing anything.
